app/controllers/events_controller.rb

```diff
@@ -219,22 +219,27 @@ def clean_events_attributes
   end
 
   def search(events, params)
+    filter = events
+    filter = filter.where( {state: params[:state]}) if params[:state].present?
+    filter = filter.where( {event_type: params[:type]}) if params[:type].present?
```

Comment:
This doesn't work: the URL parameter is called `event_type`, but here you're checking for `type`.